tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Steps/UserSteps.cs

```diff
@@ -126,15 +141,42 @@ private static string GenerateHash(string password)
             return Convert.ToBase64String(identityV3Hash.ToArray());
         }
 
-        [When(@"a GET request is made for an organisation's users with name (.*)")]
-        public async Task WhenAGETRequestIsMadeForOrganisationUsersWithName(string organisationName)
+        private string GetOrganisationIdFromName(string organisationName)
         {
             var allOrganisations = _context.Get<IDictionary<string, Guid>>(ScenarioContextKeys.OrganisationMapDictionary);
-            allOrganisations.TryGetValue(organisationName, out Guid organisationId);
+            return allOrganisations.TryGetValue(organisationName, out Guid organisationId) ? organisationId.ToString() : Guid.Empty.ToString();
+        }
 
-            using var client = new HttpClient();
-            client.SetBearerToken(_context.Get(ScenarioContextKeys.AccessToken, ""));
-            _response.Result = await client.GetAsync(new Uri($"{_organisationUrl}/{organisationId}/users"));
+        private async Task AssertUsersAreEqualToExpectedUsers(Table table, Func<EquivalencyAssertionOptions<ExpectedUserTable>, EquivalencyAssertionOptions<ExpectedUserTable>> equivalencyOptions = null)
```

Comment:
I wonder whether to split this method up. It is doing two things. The creation of the user list and the assert. I think it would help readability.